monitorcontroller/monitorcontroller.go

```diff
@@ -142,13 +142,12 @@ func (monitor *MonitorController) processQueue(ctx context.Context) {
 			monitor.Lock()
 
 			if len(monitor.monitoringQueue) > 0 && monitor.isConnected {
-				var monitoringData cloudprotocol.MonitoringData
-
-				monitoringData, monitor.monitoringQueue = monitor.monitoringQueue[0], monitor.monitoringQueue[1:]
-
 				if err := monitor.monitoringSender.SendMonitoringData(
-					monitoringData); err != nil && !errors.Is(err, amqphandler.ErrNotConnected) {
+					cloudprotocol.Monitoring{Nodes: monitor.monitoringQueue}); err != nil &&
+					!errors.Is(err, amqphandler.ErrNotConnected) {
 					log.Errorf("Can't send monitoring data: %v", err)
+				} else {
+					monitor.monitoringQueue = make([]cloudprotocol.NodeMonitoringData, 0, monitor.monitoringQueueSize)
```

Comment:
Should we clear queue in any case?

Reply:
from my point of view we have to collect data in case of failure case